### DISCUSSION TITLE
Scrollbar in long markdown tables

### DISCUSSION THREAD

**Discussion by alexeihornero** [requarks/wiki#3704] [Error / Bug Report]
I created the following test table in a .md file:

|Test1|Test2|Test3|Test4|Test5|Test6|Test7|
|-------------|-------------|---------|----------|-----------|---------|---------|
|Hello|code|code|code|code|code|longtextlongtextlongtextlongtextlonglongtextlongtextlongtextlongtextlong|

The contents of a row that expand beyond the current window are not viewable. There is also no way to scroll horizontally on the table to view the contents at the end of the row.

Any ideas how to get around this?

Thanks.

**Comment by pirvana** [comment]
The same thing happens to me, would there be any solution to this problem?

**Comment by stendall-the-one** [comment]
Same here with Firefox 71.0 (64-bit) (Ubuntu Focal)
Happens in vertical scroll too. With two panes, sometimes the left edit pane have content that it's not visible on right preview pane because (I've guess) the view it's attached to the left pane and there's no way to scroll preview pane independently.
Anyway, great project. Big thanks you for your time.

**Comment by githubdave2020** [comment]
You could add this to the CSS override in the Theme area in Administration. I'm still trying to remove padding and such.

tr:nth-child(even){background-color:#eee}tr:hover{background-color:#fff}table{width:100%;overflow-x:auto;display:block;white-space:nowrap;font-size:small;}

**Comment by alexeihornero** [comment]
Thanks, it's works for me.

**Comment by regevbr** [comment]
For me the following worked without overflow:
```css
.contents table {
  width:90%;
  overflow-x:auto;
  display:block;
  white-space:nowrap;
  font-size:small;
}
```

**Comment by nextlogic-ono** [comment]
With the current Wiki.js rendering having `figure` tag surrounding every `table` tags, this code would make all the tables at least fill the screen as well without having have to turn the `table` itself into `display: block`.

```
.contents figure {
  overflow-x: auto;
  white-space: nowrap;
}

.contents figure > table {
  width: 100%;
}
```

Additionally, having this would prevent texts such as long URLs breaking the layout (leading to breaking the popup menu position as well, which makes it hard to navigate out of the page),

```
.contents li,
.contents p {
  word-break: break-word;
}
```

I really think these should be the default as it's very easy to break the mobile experience with some long text or wide tables.

**Comment by sn4k3s** [comment]
Is there any progress with that bug without some workarounds? It kinda makes it impossible to work with large tables.

**Comment by marcelomendes** [comment]
Any workaround for this without having to hack the default theme?

**Comment by vogonsaudi** [comment]
I modified this for my use case since I wanted the contents in the table to wrap.

.contents figure {
  overflow-x: auto;
  white-space: wrap;
}

.contents figure > table {
  width: 100%;
}
.contents li,
.contents p {
  word-break: break-word;
}

I suspect this is more likely to be the the default behaviour wanted.

**Comment by pgibe** [comment]
You can also apply the format to a specific table using a custom id, e.g.:

Style:
```css
#large-table {
  overflow-x: auto;
  display: block;
  white-space: nowrap;
}
```

Page:
```
|h1|h2|h3|
|-|-|-|
|a|b|c|
{#large-table}
```
